pdm/pep517/metadata.py

```diff
@@ -315,6 +315,8 @@ def _convert_optional_dependencies(
 
     @property
     def project_name(self) -> str:
+        if self.name is None:
+            return self.filepath.parent.name
```

Comment:
Could also use `UNKNOWN` or similar.
I've also first checked if `self.filepath.name == "pyproject.toml"`, but that appears to be a given / required even?

Reply:
Use `UNKNOWN` please

Reply:
```suggestion
            return "UNKNOWN"
```